src/messages-view.tsx

```diff
@@ -81,27 +104,34 @@ export class MessagesView extends CollectionView<MessagesViewModel, MessageViewM
     );
   }
 
-  renderList({ width, height }: { width: number, height: number }) {
+  render() {
     const key = this.viewModel.channel ? this.viewModel.channel.id : null;
-    return (
-      <List
-        key={key}
-        width={width}
-        height={height}
-        deferredMeasurementCache={this.cache}
-        rowHeight={this.cache.rowHeight}
-        rowCount={this.viewModel.messagesCount}
-        rowRenderer={this.rowRenderer.bind(this)}
-      />
-    );
-  }
 
-  render() {
     return (
       <div style={{ width: '100%', height: '100%' }}>
-        <AutoSizer disableWidth>
-          {this.renderList.bind(this)}
-        </AutoSizer>
+        <InfiniteLoader
+          isRowLoaded={this.isRowLoaded.bind(this)}
+          loadMoreRows={this.loadMoreRows.bind(this)}
+          rowCount={this.viewModel.messagesCount * 2}
```

Comment:
This always needs to be bigger than the number of messages we've got cached, so that there's always room to scroll.